.github/workflows/test-integration-unreleased.yml

```diff
@@ -0,0 +1,54 @@
+name: Integration with Unreleased OpenSearch
+
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+    branches:
+      - "main"
+
+env:
+  OPENSEARCH_VERSION: '2.0'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [11, 14]
+    steps:
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+
+      - name: Checkout OpenSearch
+        uses: actions/checkout@v2
+        with:
+          repository: opensearch-project/opensearch
+          ref: ${{ env.OPENSEARCH_VERSION }}
+          path: opensearch
+
+      - name: Assemble OpenSearch and Publish to Maven Local
+        run: |
+          cd opensearch
+          ./gradlew assemble
+
+        # This step runs the docker image generated during gradle assemble in OpenSearch. It is tagged as opensearch:test.
+        # Reference: https://github.com/opensearch-project/OpenSearch/blob/2.0/distribution/docker/build.gradle#L190
+      - name: Run Docker Image
+        run: |
+          docker run -p 9200:9200 -p 9600:9600 -d -e "discovery.type=single-node" -e "bootstrap.memory_lock=true" opensearch:test
```

Comment:
I am 100% sure there is a staging repository for Docker images, @dblock @peterzhuamazon could you please refer to the right one?

Reply:
Yes there is one, but the staging image is with all the plugins which isn't ready yet. This is to test against min OpenSearch image.

Reply:
I also tried the route of writing Dockerfile and related things to create a docker image for min OpenSearch. But then I found this gradle task that runs in assemble for OpenSearch so used this one. LMK if you think the Dockerfile route is better. I can use that instead.

Reply:
I think publishing min Docker images (along with artifacts) would be the best option, may be we could quickly address that? Another possibility is to use OpenSearch test scaffolding (available as set of Gradle plugins) to run clusters from the test suites. What do you think?

Reply:
I didn't use the Gradle plugins since I wanted to create a CI way for all the clients. It can work for java client but we would still end up in the same place for other clients.

Reply:
Agree, thank you, publishing min Docker images (along with artifacts) would be the answer, right?

Reply:
Yes that would be the best option. Till that happens, is it fine if we use this change so we can start the development for 2.0? WDYT?

Reply:
I believe we will be adding API for plugins too and we need unreleased complete distribution for testing. Shall we create an issue to track this?

Reply:
Yes, please do, I think it will useful for many